charon/config.py

```diff
@@ -13,68 +13,69 @@
-
-SECTION_CHARON = "charon"
-IGNORE_PATTERN = "ignore_patterns"
-
-AWS_BUCKET = "bucket"
-AWS_DEFAULT_BUCKET = "charon"
 
+CONFIG_FILE = "charon.yaml"
 
 logger = logging.getLogger(__name__)
 
 
 class CharonConfig(object):
     """CharonConfig is used to store all configurations for charon
     tools.
-    The configuration file will be named as charon.conf, and will be stored
+    The configuration file will be named as charon.yaml, and will be stored
     in $HOME/.charon/ folder by default.
     """
-    def __init__(self, data: ConfigParser):
-        try:
-            self.__charon_configs = dict(data.items(SECTION_CHARON))
-        except NoSectionError:
-            pass
+    def __init__(self, data: Dict):
+        self.__ignore_patterns: List[str] = data.get("ignore_patterns", None)
+        self.__targets: Dict = data.get("targets", None)
+        if not self.__targets or not isinstance(self.__targets, Dict):
+            raise TypeError("Charon configuration is not correct: targets is invalid.")
 
     def get_ignore_patterns(self) -> List[str]:
-        pattern_str = self.__val_or_default(self.__charon_configs, IGNORE_PATTERN)
-        if pattern_str and pattern_str.strip() != "":
-            try:
-                return json.loads(pattern_str)
-            except (ValueError, TypeError):
-                logger.warning("Warning: ignore_patterns %s specified in "
-                               "system environment, but not a valid json "
-                               "style array. Will skip it.", pattern_str)
-        return None
-
-    def get_aws_bucket(self) -> str:
-        bucket = self.__val_or_default(self.__charon_configs, AWS_BUCKET)
+        return self.__ignore_patterns
+
+    def get_aws_bucket(self, target: str) -> str:
+        target_: Dict = self.__targets.get(target, None)
+        if not target_ or not isinstance(target_, Dict):
+            logger.error("The target %s is not found in charon configuration.")
+            return None
+        bucket = target_.get("bucket", None)
         if not bucket:
-            logger.warning("%s not defined in charon configuration,"
-                           " will use default 'charon' bucket.", AWS_BUCKET)
-            return AWS_DEFAULT_BUCKET
+            logger.error("The bucket %s is not found for target %s "
+                         "in charon configuration.")
         return bucket
 
-    def __val_or_default(self, section: dict, key: str, default=None):
-        if not section:
-            return default
-        return section[key] if section and key in section else default
+    def get_bucket_prefix(self, target: str) -> str:
+        target_: Dict = self.__targets.get(target, None)
+        if not target_ or not isinstance(target_, Dict):
+            logger.error("The target %s is not found in charon "
+                         "configuration.", target)
+            return None
+        prefix = target_.get("prefix", None)
+        if not prefix:
```

Comment:
It seems like this is a valid state. In this case, the bucket prefix path would be just `/`.

Reply:
Also, I've been through all the changes in this PR so far, and I can't find where get_bucket_prefix() is used. Am I missing it, or has this not yet been implemented?

Reply:
Yes, the prefix appending function will be used in another PR. This pr is just addressing the config changes. 
And I've set the prefix to empty if it is not specified in config in new commit.